server/proxy/service.go

```diff
@@ -38,37 +37,55 @@ type Proxy struct {
 	upstreams sync.Map // key: apps.AppID, value upstream.Upstream
 }
 
-type Service interface {
-	config.Configurable
+// Admin defines the REST API methods to manipulate Apps.
+type Admin interface {
+	DisableApp(Incoming, apps.Context, apps.AppID) (string, error)
+	EnableApp(Incoming, apps.Context, apps.AppID) (string, error)
+	InstallApp(_ Incoming, _ apps.Context, _ apps.AppID, trustedApp bool, secret string) (*apps.App, string, error)
+	UninstallApp(Incoming, apps.Context, apps.AppID) (string, error)
+}
 
-	Call(sessionID, actingUserID string, creq *apps.CallRequest) *apps.ProxyCallResponse
-	CompleteRemoteOAuth2(sessionID, actingUserID string, appID apps.AppID, urlValues map[string]interface{}) error
-	GetStatic(appID apps.AppID, path string) (io.ReadCloser, int, error)
-	GetBindings(sessionID, actingUserID string, cc *apps.Context) ([]*apps.Binding, error)
-	GetRemoteOAuth2ConnectURL(sessionID, actingUserID string, appID apps.AppID) (string, error)
-
-	Notify(cc *apps.Context, subj apps.Subject) error
-	NotifyRemoteWebhook(app *apps.App, data []byte, path string) error
-	NotifyMessageHasBeenPosted(post *model.Post, cc *apps.Context) error
-	NotifyUserHasJoinedChannel(cc *apps.Context) error
-	NotifyUserHasLeftChannel(cc *apps.Context) error
-	NotifyUserHasJoinedTeam(cc *apps.Context) error
-	NotifyUserHasLeftTeam(cc *apps.Context) error
-
-	AddLocalManifest(actingUserID string, m *apps.Manifest) (string, error)
-	AppIsEnabled(app *apps.App) bool
-	EnableApp(client mmclient.Client, sessionID string, cc *apps.Context, appID apps.AppID) (string, error)
-	DisableApp(client mmclient.Client, sessionID string, cc *apps.Context, appID apps.AppID) (string, error)
+// Invoker implements operations that invoke the Apps.
+type Invoker interface {
+	// REST API methods used by user agents (mobile, desktop, web).
+	Call(Incoming, apps.CallRequest) apps.ProxyCallResponse
+	CompleteRemoteOAuth2(_ Incoming, _ apps.AppID, urlValues map[string]interface{}) error
+	GetBindings(Incoming, apps.Context) ([]apps.Binding, error)
+	GetRemoteOAuth2ConnectURL(Incoming, apps.AppID) (string, error)
+	GetStatic(_ apps.AppID, path string) (io.ReadCloser, int, error)
+}
+
+// Notifier implements user-less notification sinks.
+type Notifier interface {
+	Notify(apps.Context, apps.Subject) error
+	NotifyRemoteWebhook(app apps.App, data []byte, path string) error
+	NotifyMessageHasBeenPosted(*model.Post, apps.Context) error
+	NotifyUserHasJoinedChannel(apps.Context) error
+	NotifyUserHasLeftChannel(apps.Context) error
+	NotifyUserHasJoinedTeam(apps.Context) error
+	NotifyUserHasLeftTeam(apps.Context) error
+}
+
+// Internal implements go API used by other packages.
+type Internal interface {
```

Comment:
Not so good name :wink: But given that I don't have a better suggestion let's keep it.

Reply:
really the most intention-revealing description I can think of is, `Mockable`? This name is used purely for documentation/grouping purposes at the moment, so perhaps it's ok to keep as is?

Reply:
Yeah, by comment is non blocking. `Mockable` doesn't say much as all interfaces are mockable.